drivers/hwinfo/hwinfo_nrf.c

```diff
@@ -63,7 +68,30 @@ ssize_t z_impl_hwinfo_get_device_id(uint8_t *buffer, size_t length)
 	return length;
 }
 
-#if !defined(CONFIG_SOC_SERIES_NRF54HX) && !defined(CONFIG_BOARD_QEMU_CORTEX_M0)
+#if RESET_CAUSE_AVAILABLE
+
+#if defined(NRF_RESETINFO)
+
+#define REASON_LOCKUP (NRFX_RESET_REASON_LOCKUP | NRFX_RESET_REASON_LOCAL_LOCKUP_MASK)
+#define REASON_SOFTWARE (NRFX_RESET_REASON_SREQ | NRFX_RESET_REASON_LOCAL_SREQ_MASK)
+#define REASON_WATCHDOG	\
+	(NRFX_RESET_REASON_DOG_MASK | \
+	 NRFX_RESET_REASON_LOCAL_DOG1_MASK | \
+	 NRFX_RESET_REASON_LOCAL_DOG0_MASK)
+
+#else /* NRF_RESETINFO */
+
+#define REASON_LOCKUP NRFX_RESET_REASON_LOCKUP_MASK
+#define REASON_SOFTWARE NRFX_RESET_REASON_SREQ_MASK
+
+#if NRF_POWER_HAS_RESETREAS
+#define REASON_WATCHDOG NRFX_RESET_REASON_DOG_MASK
+#else
+#define REASON_WATCHDOG	NRFX_RESET_REASON_DOG1_MASK
```

Comment:
Shouldn't this be actually `(NRFX_RESET_REASON_DOG0_MASK | NRFX_RESET_REASON_DOG1_MASK)`?

Reply:
Added DOG0